src/js/revert.js

```diff
@@ -3,36 +3,39 @@ add listener to the power button and send request to tab based on power on or po
 */
 
 document.addEventListener('DOMContentLoaded', function() {
-    var link = document.getElementById('link');
+
     // onClick's logic below:
-    link.addEventListener('click', function() {
-
-      browser.storage.sync.get(['powerOff'], function(result){
-
-      if(!result.powerOff){
-      browser.browserAction.setIcon({path: GREY_ICON_PATH});
-      $("#power-button").attr('src',"/img/green_button.png");
-      $("#blocked-words").css({"display":"none"});
-
-      chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
-   chrome.tabs.sendMessage(tabs[0].id, {command: SWITCH_OFF}, function(response){
-     console.log(INFO_POWER_OFF);
-   });
- });
-}else{
-
-  $("#power-button").attr('src',"/img/red_button.png");
-  $("#blocked-words").removeAttr("style");
-  chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
-chrome.tabs.sendMessage(tabs[0].id, {command: SWITCH_ON}, function(response){
- console.log(INFO_POWER_ON);
-});
-});
+    $(LINK).click(function() {
 
-}
+      browser.storage.sync.get(['power'], function(result){
 
-browser.storage.sync.set({powerOff: !result.powerOff}, function (){});
+      if(result.power){
+        browser.browserAction.setIcon({path: GREY_ICON_PATH});
+        $(POWER_BUTTON).attr(SRC,GREEN_BUTTON_PATH);
+        $(BLOCKED_WORDS).css({"display":"none"});
+
+        browser.tabs.getAllInWindow(null, function(tabs) {
+        for(var i = 0; i < tabs.length; i++){
+           browser.tabs.sendMessage(tabs[i].id, {command: SWITCH_OFF}, function(response){
+           });
+         }
+       });
+     }else{
+
```

Comment:
Should there not be a set icon here too? When you turn it back on it stays grey until you switch tabs
You can check if the icon needs to be blue/red by jQuery the blocked-num value since _revert.js_ is linked to _popup.html_